subsys/bluetooth/shell/bt.c

```diff
@@ -192,11 +201,68 @@ static bool data_cb(struct bt_data *data, void *user_data)
 	}
 }
 
+static void print_data_set(uint8_t type, uint8_t set_value_len,
+						   const uint8_t *scan_data, uint8_t scan_data_len)
+{
+	uint8_t min_value_len = MIN(set_value_len, scan_data_len);
+
+	shell_fprintf(ctx_shell, SHELL_INFO, "          Type 0x%02x: ", type);
+	for (uint8_t value_no = 0U; value_no < (scan_data_len / min_value_len); value_no++) {
+		if (value_no > 0L) {
+			shell_fprintf(ctx_shell, SHELL_INFO, ", ");
+		}
+		shell_fprintf(ctx_shell, SHELL_INFO, "0x");
+		for (uint8_t digit_no = 0U; digit_no < min_value_len; digit_no++) {
+			shell_fprintf(ctx_shell, SHELL_INFO, "%02x",
+						  scan_data[value_no*min_value_len + digit_no]);
+		}
+	}
+	shell_fprintf(ctx_shell, SHELL_INFO, "\n");
```

Comment:
Would `shell_hexdump` work as well here?

Reply:
I don't think so. The shell_hexdump seems to be made for outputting memory, including pointer offset. This will output advertising data like this:
`[DEVICE]: 40:95:78:B5:50:B9 (random), AD evt type 5, RSSI -89 FPP_HI_lakd C:1 S:0 D:0 SR:0 E:1 Prim: LE 1M, Secn: LE 2M, Interval: 0x0000 (0 us), SID: 0x0:`
`          [SCAN DATA START - EXT_ADV]`
`          Type 0x0a: 0xef`
`          Type 0x2e: 0xaf, 0x03, 0x59, 0xba, 0x2a, 0x58`
`          Type 0x16: 0x4e18, 0x01ff, 0x0fff, 0x0f00`
`          Type 0x16: 0x5318`
`          Type 0x02: 0x4d18, 0x4e18, 0x5018, 0x4418`
`          Type 0x19: 0x40, 0x0a`
`          Type 0x01: 0x06`
`          Type 0x09: FPP_HI_lakd`
`          [SCAN DATA END]`